addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -520,41 +617,34 @@ private synchronized boolean sendCubeCommand(CubeCommand command) {
      * @param terminator String with ending messagetype e.g. L:
      * @throws IOException
      */
-    private void readliness(String terminator) throws IOException {
-        if (terminator == null) {
-            return;
-        }
-        boolean cont = true;
-        while (cont) {
+    private void readLines(String terminator) throws IOException {
+        while (true) {
             String raw = reader.readLine();
-            if (raw != null) {
-                logger.trace("message block: '{}'", raw);
-                try {
-                    this.messageProcessor.addReceivedLine(raw);
-                    if (this.messageProcessor.isMessageAvailable()) {
-                        Message message = this.messageProcessor.pull();
-                        processMessage(message);
-
-                    }
-                } catch (UnprocessableMessageException e) {
-                    if (raw.contentEquals("M:")) {
-                        logger.info("No Rooms information found. Configure your MAX! Cube: {}", ipAddress);
-                        this.messageProcessor.reset();
-                    } else {
-                        logger.info("Message could not be processed: '{}' from MAX! Cube lan gateway: {}:", raw,
-                                ipAddress);
-                        this.messageProcessor.reset();
-                    }
-                } catch (Exception e) {
-                    logger.info("Error while handling message block: '{}' from MAX! Cube lan gateway: {}:", raw,
-                            ipAddress, e.getMessage(), e);
-                    this.messageProcessor.reset();
+            if (raw == null) {
+                return;
+            }
+            logger.trace("message block: '{}'", raw);
+            try {
+                this.messageProcessor.addReceivedLine(raw);
+                if (this.messageProcessor.isMessageAvailable()) {
+                    Message message = this.messageProcessor.pull();
+                    processMessage(message);
                 }
-                if (terminator == null || raw.startsWith(terminator)) {
-                    cont = false;
+            } catch (UnprocessableMessageException e) {
+                if (raw.contentEquals("M:")) {
+                    logger.info("No Rooms information found. Configure your MAX! Cube: {}", ipAddress);
+                    this.messageProcessor.reset();
+                } else {
+                    logger.info("Message could not be processed: '{}' from MAX! Cube lan gateway: {}:", raw, ipAddress);
+                    this.messageProcessor.reset();
                 }
-            } else {
-                cont = false;
+            } catch (Exception e) {
```

Comment:
I agree. I couldn't clean this because I didn't know which exception was actually the possible culprit.